pkg/subctl/cmd/cloud/utils/aws.go

```diff
@@ -0,0 +1,100 @@
+/*
+© 2021 Red Hat, Inc. and others.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	"errors"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/submariner-io/cloud-prepare/pkg/api"
+	cloudprepareaws "github.com/submariner-io/cloud-prepare/pkg/aws"
+	"github.com/submariner-io/submariner-operator/pkg/subctl/cmd/utils"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+// RunOnAWS runs the given function on AWS, supplying it with a cloud instance connected to AWS and a reporter that writes to CLI.
+// The functions makes sure that infraID and region are specified, and extracts the credentials from a secret in order to connect to AWS.
+func RunOnAWS(infraID, region, gwInstanceType, kubeConfig, kubeContext string,
+	function func(cloud api.Cloud, reporter api.Reporter) error) error {
+	if infraID == "" {
+		utils.ExitWithErrorMsg("You must specify the infraid")
+	}
+
+	if region == "" {
+		utils.ExitWithErrorMsg("You must specify the region")
+	}
+
+	k8sConfig, err := utils.GetRestConfig(kubeConfig, kubeContext)
+	utils.ExitOnError("Filed to initialize a Kubernetes config", err)
+
+	reporter := NewCLIReporter()
+	reporter.Started("Retrieving AWS credentials from your OpenShift installation")
+	creds, err := getAWSCredentials(k8sConfig)
```

Comment:
I think it would be cool to also let us use the local credentials on the .aws dir (so no tweaking necessary of the permissions), but we can improve that during 0.10

Reply:
yes we can always add more handling later